tests/Export/ProductServiceTest.php

```diff
@@ -237,4 +238,97 @@ public function testVariantsWithSameParentButDifferentDisplayGroupAreExportedAsS
         $this->assertSame($expectedFirstVariantId, $products[0]->getId());
         $this->assertSame($expectedSecondVariantId, $products[1]->getId());
     }
+
+    public function testTheMainProductIsBasedOnTheMainVariantIdOfADisplayGroup(): void
+    {
+        $expectedParentId = Uuid::randomHex();
+        $expectedFirstVariantId = Uuid::randomHex();
+        $expectedSecondVariantId = Uuid::randomHex();
+        $expectedMainVariantId = $expectedSecondVariantId;
+
+        $firstOptionId = Uuid::randomHex();
+        $secondOptionId = Uuid::randomHex();
+        $optionGroupId = Uuid::randomHex();
+
+        $this->createVisibleTestProduct([
+            'id' => $expectedParentId,
+            'active' => false,
+            'configuratorSettings' => [
+                [
+                    'option' => [
+                        'id' => $firstOptionId,
+                        'name' => 'Red',
+                        'group' => [
+                            'id' => $optionGroupId,
+                            'name' => 'Color',
+                        ],
+                    ],
+                ],
+                [
+                    'option' => [
+                        'id' => $secondOptionId,
+                        'name' => 'Orange',
+                        'group' => [
+                            'id' => $optionGroupId,
+                            'name' => 'Color',
+                        ],
+                    ],
+                ],
+            ],
+            'configuratorGroupConfig' => [
+                [
+                    'id' => $optionGroupId,
+                    // Explicitly set this to false. This tells Shopware to consider the mainVariationId (if set).
+                    'expressionForListings' => false,
```

Comment:
For my understanding, what does expressionForListings mean?

Reply:
It depends on if the variant should be split or not.
![image](https://user-images.githubusercontent.com/20555112/142393840-55c597c6-b5cb-410a-a203-ec3eed68b6dc.png)